001-core/constants.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Automattic\VIP\Core\Constants;
+
+/**
+ * Define DB_HOST, DB_USER, DB_PASSWORD, and DB_NAME constants.
+ *
+ * WPVIP uses use HyperDB, so DB_* constants are not automatically defined.
+ * As a best practice, these constants should not be used directly, but we define them here in
+ * case any customer code does to avoid fatal errors in PHP 8+.
+ *
+ * @param object $wpdb
+ */
+function define_db_constants( $wpdb ): void {
+	if ( defined( 'DB_NAME' ) || defined( 'DB_HOST' ) || defined( 'DB_PASSWORD' ) || defined( 'DB_USER' ) ) {
+		return;
+	}
+
+	if ( ! method_exists( $wpdb, 'get_hyper_servers' ) ) {
+		return;
+	}
+
+	$db_servers = $wpdb->get_hyper_servers( 'write' );
+	if ( ! is_array( $db_servers ) ) {
+		return;
+	}
+
+	if ( isset( $db_servers[1] ) ) {
+		// The index is based on the "write priority", which defaults to 1.
+		// See https://github.com/Automattic/HyperDB/blob/06dc2449535c35aaaf2f7e1e8a28fb40e59ff23e/db.php#L257-L267
+		$db = $db_servers[1];
+	} else {
+		$keys = array_keys( $db_servers );
+		$db   = $db_servers[ $keys[0] ] ?? null;
+	}
```

Comment:
Minor but I think the better approach here would be to sort by priority and pull the one with the highest value / index:  

https://github.com/Automattic/HyperDB/blob/06dc2449535c35aaaf2f7e1e8a28fb40e59ff23e/db.php#L558-L559

Not sure if that impacts performance though.

Reply:
I don't think a sort on an array of < 10 elements would be that big of a deal